File: QT Creator/Footballer.cpp
#include "Footballer.h"

using namespace std;

BOOST_CLASS_VERSION(Footballer, 1)
BOOST_CLASS_EXPORT(Footballer)


QStringList Footballer::get_data() const
{
    return Sportsman::get_data() + QStringList{QString::number(number),
                                               QString::fromLocal8Bit(position.c_str())};
}

std::shared_ptr<Sportsman> Footballer::clone() const{
    return std::make_shared<Footballer>(*this);
}

void Footballer::update(QStringList &data)
{
    Sportsman::update(data);
    number = data[4].toInt();
    position = data[5].toLocal8Bit();
}

File: С#_Desktop_Application/C#_Form/MyAppIlichev/Footballer.cs
using System;

namespace MyAppIlichev
{
    internal class Footballer : Sportsman
    {
        public string Position { get; set; }
        public int Number { get; set; }

        public Footballer()
            : base()
        {
            Position = "";
            Number = 0;
        }

        public override SportsmanDataStruct GetStruct() 
        {
            SportsmanDataStruct SD = base.GetStruct();
            SD.number = Number;
            SD.position = Position;

            return SD;
        }

        public override List<string> GetData()
        {
            List<string> data = base.GetData();
            data.Add(Number.ToString());
            data.Add(Position);

            return data;
        }

    }
}
